### PR TITLE
db, add support for java

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

**Comment by Begasus** [comment]
@jmairboeck created a draft for this, the library and the jar files are included, but inside the build I see a lot of errors like:
```
../util/db_gui/datapage/DataAccessController.java:18: error: package javafx.fxml does not exist
import javafx.fxml.FXML;
                  ^
../util/db_gui/datapage/DataAccessController.java:19: error: package javafx.scene.control does not exist
import javafx.scene.control.TextArea;
                           ^
../util/db_gui/datapage/DataAccessController.java:29: error: cannot find symbol
    private TextArea FeedbackBox;
            ^
  symbol:   class TextArea
  location: class DataAccessController
../util/db_gui/datapage/QueueController.java:21: error: package javafx.event does not exist
import javafx.event.ActionEvent;
                   ^
../util/db_gui/datapage/QueueController.java:22: error: package javafx.fxml does not exist
import javafx.fxml.FXML;
                  ^
../util/db_gui/datapage/QueueController.java:23: error: package javafx.scene.control does not exist
import javafx.scene.control.ComboBox;
                           ^
```

I'm probably missing something in the recipe, maybe you could shed a light on this? ;)

**Comment by jmairboeck** [comment]
Sorry, I don't really know anything about JavaFX. (It is about 10 years since I did something with Java really, so I'm a bit rusty there and don't really know the modern stuff.)

Probably JavaFX isn't included in our java packages. I think modern versions of Java usually include it on other platforms. Before, it was a separate library. It is a GUI library and would probably need a proper Haiku implementation to get it working (or something based on xlibe or wayland).

I think the java part of db is probably not really worth it.

File: dev-db/db/db18-18.1.40.recipe [hunk 1]
```diff
@@ -65,36 +62,66 @@ PROVIDES_devel="
...
+		"
+fi
+
 BUILD()
 {
-#	export CPPFLAGS="-I/boot/system/lib/openjdk17/include/haiku"
+	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+		source /system/data/profile.d/openjdk.sh
+		export CPPFLAGS="-I/boot/system/lib/openjdk17/include/haiku"
+		java="--enable-gui --enable-java"
```
Comment (jmairboeck) [inline dev-db/db/db18-18.1.40.recipe:119]: You could leave out the GUI and provide just the Java bindings part.

Comment (Begasus) [reply dev-db/db/db18-18.1.40.recipe:119]: It was worth a try :) Will do some check tomorrow on 32bit for this. Thanks so far!

**Comment by korli** [comment]
I think JavaFX needs native support.

**Comment by Begasus** [comment]
Should be good to go, doing a final test build on 32bit secondary architecture.

**Pull request merged**